data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/SqsWorker.java

```diff
@@ -247,6 +260,10 @@ private Optional<DeleteMessageBatchRequestEntry> processS3Object(
             final AcknowledgementSet acknowledgementSet) {
         // SQS messages won't be deleted if we are unable to process S3Objects because of S3Exception: Access Denied
         try {
+
+            if (Objects.nonNull(sqsDynamicVisibilityTimeoutManager)) {
+                sqsDynamicVisibilityTimeoutManager.startDynamicVisibilityForMessage(parsedMessage.message, s3Service.getObjectSize(s3ObjectReference));
```

Comment:
We need to call stopDynamic in negative scenarios, say if there's a retryable exception in downloading the object

Basically any time the processing of the object fails, we should kill the dynamic viz thread

Reply:
Makes sense. I'll make a `stopDynamic` call in the catch block of this method. Are there any other locations where this is needed?

Reply:
I'm not sure. If there's a single top-level try/catch around the S3 processing logic then that should be the only place it's necessary